src/safe_allocator.h

```diff
@@ -68,6 +69,8 @@ class SafeAllocator : public Allocator {
 private:
     Allocator* const raw_allocator_ = nullptr;
     std::shared_ptr<Allocator> owned_allocator_ = nullptr;
+
+    bool owner_{false};
```

Comment:
need deallocator to manage raw_allocator ?

Reply:
ditto